src/components/Navbar/Navbar.jsx

```diff
@@ -0,0 +1,125 @@
+import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import { convertKelvinToCelsius } from '../../utils/convertTemperatures'
+
+import useFetchData from '../../hooks/useFetchData'
+
+import Box from '../Box/Box'
+import ImageIcon from '../ImageIcon/ImageIcon'
+
+import './Navbar.style.scss'
+
+import iconLocationLight from '../../assets/icon-location-dark.png'
+import iconLocationDark from '../../assets/icon-location-light.png'
+import iconThemeSun from '../../assets/icon-theme-sun-light.png'
+import iconThemeMoon from '../../assets/icon-theme-moon-dark.png'
+import iconSettingLight from '../../assets/icon-settings-dark.png'
+import iconSettingDark from '../../assets/icon-settings-light.png'
+import iconGeolocationLight from '../../assets/icon-geolocation-dark.png'
+import iconGeolocationDark from '../../assets/icon-geolocation-light.png'
+
+import currentWeatherDataAPI from '/public/data.json'
+
+export default function Navbar({ onThemeToggle, currentTheme, location, requestGeolocation, setCurrentWeatherData}) {
```

Comment:
imo instead of passing setCurrentWeatherData you should just pass location details as props if its needed in one of the navigation items.
also, requestGeolocation is not really concise with 'settings button'. this component should probably accept event handler onSettingsButtonClick instead